### ISSUE TITLE
apps.nextcloud.com server unreliable

### ISSUE THREAD

**Issue by chaotika** [nextcloud/appstore#761]
Dear Nextcloud Developers,

i am working on a project to automate the installation of a nextcloud instance for a research project. We currently struggle with unreliable servers on apps.nextcloud.com and download.nextcloud.com. For the download-Server i already configured it to use the github mirror but also apps.nextcloud.com is regularily unavailable, which makes it hard to reliably use nextcloud with our CI/CD deployment.

We plan to publish our code to the open source community after finishing our project in which nextcloud is an important component. We would hope that we can reliably deploy it.

Thank you for providing nextcloud, i think that this is a great project and i like the new features of nextcloud 20!

Here is an exemplary error that might occur in my ansible script:

```
 [get latest nextcloud version] 
fatal: [cloud]: FAILED! => changed=false 
  elapsed: 30
  msg: 'Status code was -1 and not [200]: Request failed: <urlopen error [Errno 101] Network is unreachable>'
  redirected: false
  status: -1
  url: https://apps.nextcloud.com/api/v1/platforms.json
```

**Comment by chaotika** [comment]
Possible solutions:
- ask community for mirrors
- commercial CDN provider
- sourceforge.net?

**Comment by SeaLife** [comment]
Hey,

we also cant access the "Apps" page in our nextcloud instance probably due to this issue...

**Comment by pm-pm** [comment]
This is not the point.
It is NOT interessing that we can´t access the server.
The questions is: "When will it work again?" 10 Minutes? 2 Hours? Tomorrow?

**Comment by christ** [comment]
I suggest that the page not rely on apps.nextcloud.com to be available in order to load your apps page.

**Comment by pm-pm** [comment]
It is working now again. :-)

**Comment by flufftronix** [comment]
I was getting "host unreachable" earlier when trying to load up the apps page in my nextcloud install, and eventually figured out this was happening because the domain was resolving to an ipv6 address, which my ISP doesn't use.

And then of course, right as I finish setting up [Tunnelbroker](https://www.tunnelbroker.net/) across my network, Pihole, VM host and Docker.. ipv4 pings to apps.nextcloud.com start coming back. You're welcome, everyone! XD

**Comment by SeaLife** [comment]
I cant believe it, is there a way to mirror the appstore and use a mirrored version?

The store is again nearly unreachable..

**Comment by chaptergy** [comment]
I wasn't able to open the apps page since it kept loading. However after a few minutes it does seem to at least load the currently installed plugins. Maybe detatching the loading of the currently installed apps from loading of new apps would improve this, so we can always manage the currently installed apps right away without having to wait for a timeout. Also maybe an info message, that connecting to the server is taking longer than expected would tell the user what is happening.

**Comment by Shyru** [comment]
I have the same problem, just installed a new nextcloud instance using docker and nginx as proxy and I'm constantly running into Gateway Timeout errors from nginx and thought I did something wrong with the setup. I'm also not able to load the System-Settings. Problem is: I need the server urgently. Is there any workaround? Proxy? Can the app-store-integration be disabled? Can I install apps manually without the app-store?

**Comment by MarslMarcello** [comment]
Same here app.nextcloud.com not available. Does somebody know what is/was happening and when can we expect it will be back. I was a bit concerned because I just updated my Nextcloud to version 20.0 and thought I did something wrong.
Thanks

**Comment by borzel** [comment]
Same here... :-(

**Comment by BORIKKEN21** [comment]
Still down :(

**Comment by armaccloud** [comment]
Down here too

**Comment by Sandman0984** [comment]
Still down.  I updated to NC20 (manually) a few hours ago without realizing the app store was down.  The update succeeded, but most of my critical apps (calendar, contacts, talk, and more) were marked as disabled/untested.  The app page was also missing all access to the app store in the menu.  It wasn't until looking into this that I found "apps.nextcloud.com" was down. Enabling the apps allowed me to access my data, but the apps are still out of date.

**Comment by kesselb** [comment]
@chaotika

I would like to suggest a different way for your CI/CD use case. Your pipeline should not break because a external service is unavailable. But most of the time you have no control about an external service. 

For critical pipelines we use tools like satis (https://getcomposer.org/doc/articles/handling-private-packages-with-satis.md) to create a static repository for our internal packages and external libraries. If an external service is unavailable we still have a local copy. This copy might be older but most of the time that's better than no copy.

Back to Nextcloud: You can setup something similar for the apps store. How to use a self hosted apps store is explained here https://docs.nextcloud.com/server/20/admin_manual/apps_management.html#using-a-self-hosted-apps-store.

The apps store needs https://apps.nextcloud.com/api/v1/categories.json and https://apps.nextcloud.com/api/v1/apps.json. With some lines of Bash or PHP it should be possible to mirror those files and use it as custom apps store. At best you have some monitoring the make sure the update / mirror process works.

**Comment by kowalcj0** [comment]
My `investigator` 🐊 said `Connection timed out`  😞

```
curl -vi https://apps.nextcloud.com/
*   Trying 176.9.217.53...
* TCP_NODELAY set
*   Trying 2a01:4f8:130:32f1::53...
* TCP_NODELAY set
* Immediate connect fail for 2a01:4f8:130:32f1::53: Network is unreachable
*   Trying 2a01:4f8:130:32f1::53...
...

* Failed to connect to apps.nextcloud.com port 443: Connection timed out
* Closing connection 0
curl: (7) Failed to connect to apps.nextcloud.com port 443: Connection timed out
```

**Comment by mauri5hut** [comment]
Same problem here, the page isn't available in a normal webbrowser... what's going on?

**Comment by strugee** [comment]
All, please refrain from posting "me too" comments. You're not adding anything to the discussion since by now it's clear that apps.nextcloud.com is down globally, and every time you do, you send email to at least the 38 people watching this repository, plus anyone subscribed to this issue (like myself). At this point we'll just have to wait for someone involved in the app store to fix the issue and generating more email for them to wade through isn't going to help. Thanks.

**Comment by SeaLife** [comment]
I wrote a simple FastAPI service which (for now) caches those json files @kesselb mentioned *indefinitely*.

My Apps page is now fast as hell again :D

There is a Git repository u can check out if u want https://git.r3ktm8.de/SeaLife-Docker/nextcloud-cache
U can also use the docker image at https://git.r3ktm8.de/SeaLife-Docker/nextcloud-cache/container_registry/

I've added those parameters to my config.php

```php
$config = array(
  'appstoreenabled' => true,
  'appstoreurl' => 'https://apps.nextcloud.example.com/api/v1',
)
```

I hope it can help some of you guys..

A small hint: The files will be initialy fetched from the appstore, so the store must be reachable once.. But u can just look at the kind to get know what the service is doing :P

**Comment by mielleman** [comment]
I also cannot reach the https://apps.nextcloud.com/api/v1/platforms.json url from my Nextcloud instance.
From **three** other internet connections I can access the url using the same TLSv1.2 curl settings **without any problems**.

Of course I checked if the router or my firewall was the problem, bus also with those turned off I could not reach the https://apps.nextcloud.com/api/v1/platforms.json url.

Any tips on what I should check/try/do?

**Comment by mielleman** [comment]
If I have to guess, I am being blocked on the apps.nextcloud.com instance due to something like fail2ban to prevent ddos. Since a "curl https://apps.nextcloud.com/api/v1/apps.json" is giving you a nice request:response ratio ...

**Comment by PetrVladimirov** [comment]
apps.nextcloud.com is down again... 

![image](https://user-images.githubusercontent.com/35421649/106353885-d7670400-62fe-11eb-9089-cbde36bc7bc5.png)

**Comment by nunesgh** [comment]
Since a couple days ago, my Nextcloud instance has been unable to access the appstore. But according to the logs, the 'server version is invalid'. Is this a result of app.nextcloud.com server being unreliable?

Here follows the log:

> [appstoreFetcher] Warning: InvalidArgumentException: server version is invalid at <<closure>>
> 
>  0. /var/www/nextcloud/lib/private/App/AppStore/Fetcher/AppFetcher.php line 119
>     OC\App\CompareVersion->isCompatible("7.3.19-1~deb10u1", "5.6", ">=")
>  1. /var/www/nextcloud/lib/private/App/AppStore/Fetcher/Fetcher.php line 187
>     OC\App\AppStore\Fetcher\AppFetcher->fetch("", "", false)
>  2. /var/www/nextcloud/lib/private/Installer.php line 411
>     OC\App\AppStore\Fetcher\Fetcher->get(false)
>  3. /var/www/nextcloud/apps/settings/lib/Controller/AppSettingsController.php line 153
>     OC\Installer->isUpdateAvailable("files_automatedtagging")
>  4. /var/www/nextcloud/apps/settings/lib/Controller/AppSettingsController.php line 136
>     OCA\Settings\Controller\AppSettingsController->getAppsWithUpdates()
>  5. /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 169
>     OCA\Settings\Controller\AppSettingsController->viewApps()
>  6. /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 100
>     OC\AppFramework\Http\Dispatcher->executeController(OCA\Settings\Con ... {}, "viewApps")
>  7. /var/www/nextcloud/lib/private/AppFramework/App.php line 152
>     OC\AppFramework\Http\Dispatcher->dispatch(OCA\Settings\Con ... {}, "viewApps")
>  8. /var/www/nextcloud/lib/private/Route/Router.php line 309
>     OC\AppFramework\App::main("OCA\\Settings\\ ... r", "viewApps", OC\AppFramework\ ... {}, {category: "",id ... "})
>  9. /var/www/nextcloud/lib/base.php line 1008
>     OC\Route\Router->match("/settings/apps")
> 10. /var/www/nextcloud/index.php line 37
>     OC::handleRequest()

**Comment by SeaLife** [comment]
Hey @nunesgh ,

take a look at my comment https://github.com/nextcloud/appstore/issues/761#issuecomment-706764287 .

I've experienced no problems at all using a cached version of the app store.
The nightly-build for yesterday failed because the shop is not reachable, but the nightly build from 2 days ago was OK, so the cache in the container is 2 days old. :)
(https://git.r3ktm8.de/SeaLife-Docker/nextcloud-cache/-/pipelines)

Regards
SeaLife

**Comment by pravardhanreddy** [comment]
I installed nextcloud 20.0.6 on rpi4 today. Facing the same issue. Please fix.

**Comment by bigcat88** [comment]
AppStore should become faster from now(https://github.com/nextcloud/appstore/pull/1147), feel free to check and open a new issue with suggestions if any.

_Just be careful, in AppStore there is still limit on how many apps.json endpoint calls you can do in a period of time without receiving a ban. _

I'll close this issue as it is old, and do not reflect all changes performed in last few months.
